src/backend/Wayland.cpp

```diff
@@ -769,8 +768,13 @@ void Aquamarine::CWaylandOutput::scheduleFrame(const scheduleFrameReason reason)
 
     if (waylandState.frameCallback)
         frameScheduledWhileWaiting = true;
-    else
-        backend->idleCallbacks.emplace_back([this]() { sendFrameAndSetCallback(); });
+    else {
+        auto w = self;
+        backend->idleCallbacks.emplace_back([w]() {
```

Comment:
actually you dont need that line above, you can do `[w = self]` it should do the same, though I am not 100% sure if it will use a & or not. Worth checking

Reply:
It copies as it should. I wasn't aware of that syntax when I wrote this but it works well